Tools/TestWebKitAPI/Tests/WebKit/WKBackForwardListTests.mm

```diff
@@ -808,3 +810,50 @@ static void runBackForwardNavigationDoesNotSkipItemsWithUserGestureTest(Function
 
     EXPECT_STREQ([webView URL].absoluteString.UTF8String, url1.absoluteString.UTF8String);
 }
+
+TEST(WKBackForwardList, BackForwardListRemoveAndAddSubframes)
+{
+    auto indexHTML = "<iframe id='1' src='/frame1'></iframe>"
+        "<iframe id='2' src='/frame2'></iframe>"_s;
+    TestWebKitAPI::HTTPServer server({
+        { "/index"_s, { indexHTML } },
+        { "/frame1"_s, { ""_s } },
+        { "/frame2"_s, { "<script> alert('done'); </script>"_s } },
```

Comment:
It would be nice if these two alerts were different so we could tell from the test below which frame was done loading.